### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by Zaista**
What will this change do: 

- Move all tests to `tests` folder in order to have a clear separation of project, src, unit and regression testing files
- Fixed an error on windows machines with triggered line-ending git changes to constantly show up
- Fixed missing optional declaration for `commandOptions` in `command.ts` file
- Refactored unit tests to:
  - remove all generated files/directories after successful test run
  - moved paths from global to local scope easier debugging
- changed error handling in `image.ts` file to throw an actuall error (not to assume that file doesn't exists)
- added webpack preprocessor in `cypress.common.config.ts` configuration file to enable typescript aliases for easier imports
- refactored all regression tests to import the `cypress-visual-regression` project instead of importing directly from `src` folder to make plugin tests more realistic

### CODE REVIEW

File: src/plugin.spec.ts [hunk 1]
```diff
@@ -10,22 +10,11 @@ import {
...
-    } // If the error is "File not found," do nothing (fail silently)
-  }
-}
-
 const baseUpdateOptions: UpdateSnapshotOptions = {
   screenshotName: 'enjuto',
   specName: 'sub-folder',
-  screenshotAbsolutePath: path.join('fixtures', 'assets', 'base', 'enjuto.png'),
-  baseDirectory: path.join('fixtures', 'assets', 'base'),
+  screenshotAbsolutePath: path.join('tests/integration/cypress/fixtures/assets/base/enjuto.png'),
```
Comment (area73) [inline src/plugin.spec.ts:16]: Is this a good idea?
I have my doubts about that this will work in all OS.

Comment (Zaista) [reply src/plugin.spec.ts:16]: This is what I wanted to put to test with the latest platform checks, but then I started getting a really weird problems with .png images getting unrecognized, could not find my way around it still

Comment (Zaista) [reply src/plugin.spec.ts]: Ok, seems to be working just fine on all platforms, so I'd say it's ready for another review @area73

File: tests/integration/cypress/web/02.html [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" />
   </head>
   <body>
     <div class="container">
       <div class="row">
         <div class="col-xs-4">
           <h1>Register</h1>
-          <hr><br>
+          <hr />
+          <br />
```
Comment (area73) [inline tests/integration/cypress/web/02.html:14]: I guess all these auto close tags were made by the linter. I’m fine with it but just to let you know that they are not needed

**Review by area73 — APPROVED** [review]
🚀, nice job 👍.
I specially like the simplification of the filepaths and also about using rmsync (I think that was a hot spot in some failing test)

File: src/plugin.spec.ts (hunk 1 above)
Comment (area73) [reply src/plugin.spec.ts]: Done !

**Pull request merged**